freetextresponse/freetextresponse.py

```diff
@@ -0,0 +1,486 @@
+            ('FreeText XBlock',
+             '''<sequence_demo>
+                    <freetextresponse />
+                    <freetextresponse name='My First XBlock' />
+                </sequence_demo>
+             '''),
+        ]
+        return scenarios
+
+    display_correctness = Boolean(
+        display_name=_('Display Correctness?'),
+        help=_(
+            'This is a flag that indicates if the indicator '
+            'icon should be displayed after a student enters '
+            'their response'
+        ),
+        default=True,
+        scope=Scope.settings,
+    )
+    display_name = String(
+        display_name=_('Display Name'),
+        help=_(
+            'This is the title for this question type'
+        ),
+        default=_('Free-text Response'),
+        scope=Scope.settings,
+    )
+    fullcredit_keyphrases = List(
+        display_name=_('Full-Credit Key Phrases'),
+        help=_(
+            'This is a list of words or phrases, one of '
+            'which must be present in order for the student\'s answer '
+            'to receive full credit'
+        ),
+        default=[],
+        scope=Scope.settings,
+    )
+    halfcredit_keyphrases = List(
+        display_name=_('Half-Credit Key Phrases'),
+        help=_(
+            'This is a list of words or phrases, one of '
+            'which must be present in order for the student\'s answer '
+            'to receive half credit'
+        ),
+        default=[],
+        scope=Scope.settings,
+    )
+    max_attempts = Integer(
+        display_name=_('Maximum Number of Attempts'),
+        help=_(
+            'This is the maximum number of times a '
+            'student is allowed to attempt the problem'
+        ),
+        default=0,
+        values={'min': 1},
+        scope=Scope.settings,
+    )
+    max_word_count = Integer(
+        display_name=_('Maximum Word Count'),
+        help=_(
+            'This is the maximum number of words allowed for this '
+            'question'
+        ),
+        default=10000,
+        values={'min': 1},
+        scope=Scope.settings,
+    )
+    min_word_count = Integer(
+        display_name=_('Minimum Word Count'),
+        help=_(
+            'This is the minimum number of words required '
+            'for this question'
+        ),
+        default=0,
+        values={'min': 1},
+        scope=Scope.settings,
+    )
+    prompt = String(
+        display_name=_('Prompt'),
+        help=_(
+            'This is the prompt students will see when '
+            'asked to enter their response'
+        ),
+        default=_('Please enter your response within this text area'),
+        scope=Scope.settings,
+    )
+    weight = Integer(
+        display_name=_('Weight'),
+        help=_(
+            'This assigns an integer value representing '
+            'the weight of this problem'
+        ),
+        default=1,
+        values={'min': 1},
+        scope=Scope.settings,
+    )
+
```

Comment:
Actually the spacing is intentional. I grouped all Scope.settings variables together in alphabetical order (without spaces in between), and did the same for all Scope.user_state variables.